lib/types.nix

```
@@ -80,10 +80,42 @@ rec {
   };
 
 
+  /**
+    Merges two option types. The order of the two types should be irrelevant.
+
+    It uses their internal implementation given in typeMerge,
+    but in case the first type cannot merge itself with the second,
+    it also tests if the second type can merge itself with the first,
+    making this operation commutative.
+
+    # Example
+
+    ```nix
+    mergeTypes (enum [ "first" ]) (enum [ "second" ])
+    ```
+
+    # Type
+
+    ```
+    mergeTypes :: AttrSet -> AttrSet -> AttrSet
+    ```
+
+    # Arguments
+
+    - [t1] first type to merge
+    - [t2] second type to merge
```

While this isn't getting rendered just yet, let's make it already adhere to the standard section structure, see e.g. [here](https://github.com/NixOS/nixpkgs/blob/dd3a11f70d293a5330db4bf104325423e4daa630/lib/attrsets.nix#L17)

```suggestion
    # Arguments
    
    `t1`
    
    : first type to merge
    
    `t2`
    
    : second type to merge
    
    # Type

    ```
    mergeTypes :: AttrSet -> AttrSet -> AttrSet
    ```

    # Examples
    :::{.example}
    ## `lib.types.mergeTypes` usage example
    ```nix
    mergeTypes (enum [ "first" ]) (enum [ "second" ])
    => enum [ "first" "second" ]
    ```
    :::
```